elastic/kibana
# [Logs Stream] Support date nanos

closes #88290 

## 📝  Summary

As described in #88290 we need to add `date_nanos` support to the Stream UI page. In this PR the necessary changes have been made all over the Stream UI and the usages of it. 

## ✅  Testing

⚠️ Testing the Stream UI with old timestamp indices is important to make sure that the behavior is still as before and not affected at all. This can be done by running local env from the PR and simulating all interactions on edge-lit cluster for example, to make sure that the behavior is not changed.

For testing the new changes with support of `date_nano`:

1. You can use [the steps here](https://github.com/elastic/kibana/issues/88290#issuecomment-1713400873) to create and ingest documents with nano precision.
2. Navigate to the stream UI and the documents should be displayed properly.
3. Sync with the URL state should be changed from timestamp to ISO string date.
4. Changing time ranges should behave as before, as well as Text Highlights.
5. Open the logs flyout and you should see the timestamp in nano seconds.
6. Play around with the minimap, it should behave exactly as before.

### Stream UI:

<img width="2556" alt="Screenshot 2023-11-02 at 14 15 49" src="https://github.com/elastic/kibana/assets/11225826/596966cd-0ee0-44ee-ba15-f387f3725f66">

- The stream UI has been affected in many areas:
    -  The logPosition key in the URL should now be in ISO string, but still backward compatible incase the user has bookmarks with timestamps.
    - The minimap should still behave as before in terms of navigation onClick and also highlighting displayed areas

### Stream UI Flyout:

<img width="2556" alt="Screenshot 2023-11-02 at 14 15 23" src="https://github.com/elastic/kibana/assets/11225826/6081533c-3bed-43e1-872d-c83fe78ab436">

- The logs flyout should now display the date in nanos format if the document is ingested using a nano format.

### Anomalies:

<img width="1717" alt="Screenshot 2023-11-01 at 10 37 22" src="https://github.com/elastic/kibana/assets/11225826/b6170d76-40a4-44db-85de-d8ae852bc17e">

-Anomalies tab under logs as a navigation to stream UI which should still behave as before passing the filtration and time.

### Categories:

<img width="1705" alt="Screenshot 2023-11-01 at 10 38 19" src="https://github.com/elastic/kibana/assets/11225826/c4c19202-d27f-410f-b94d-80507542c775">

-Categories tab under logs as a navigation to stream UI which should still behave as before passing the filtration and time.

### External Links To Stream:
- All links to the Stream UI should still work as before:
   - APM Links for traces, containers, hosts
   - Infra links in Inventory and Hosts views

## 🎥 Demo

https://github.com/elastic/kibana/assets/11225826/9a39bc5a-ba37-49e0-b7f2-e73260fb01f0
